src/threads_arg/serialization.py

```diff
@@ -74,7 +74,7 @@ def serialize_instructions(instructions, out, variant_metadata=None, allele_ages
     dset_het_s[:] = flat_mismatches
     dset_range[:] = [region_start, region_end]
 
-    if variant_metadata is not None:
+    if (variant_metadata is not None) and (num_sites > 0):
```

Comment:
Optional tidy-up:
The original `is not None` is not needed so this could be written
```python
if variant_metadata and (num_sites > 0):
```
but also - if you were happy to disregard the case for num_sites being less than zero - this could be even more succintly written as
```python
if variant_metadata and num_sites:
```
... becasue python treats equals zero and None checks the same way.
What do you think?